src/algebra/star/order.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2023 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+
+import algebra.star.basic
+import group_theory.submonoid.basic
+
+/-! # Star ordered rings
+
+We define the class `star_ordered_ring R`, which says that the order on `R` respects the
+star operation, i.e. an element `r` is nonnegative iff it is in the `add_submonoid` generated by
+elements of the form `star s * s`. In many cases, including all C⋆-algebras, this can be reduced to
+`0 ≤ r ↔ ∃ s, r = star s * s`. However, this generality is slightly more convenient (e.g., it
```

Comment:
Can we add a lemma that states this (and perhaps use it to construct the `real` and `complex` instances, if that turns out to make them shorter)?

Reply:
I have added these constructors and used it to essentially restore the previous `real` and `complex` instances. The diff is now fairly minimal in regards to those.